main.cpp

```diff
@@ -1,479 +1,535 @@
-    can use 'std::string' to represent strings in this project.
- */
+    It does not make sense to pass in a new Manager whenever you are going to train your team players.
+    Soccer teams have ONE manager.
+
+    a much more relevant usage would be adding a member function that hires a new manager:
+
+    struct SoccerTeam
+    {
+        struct Manager
+        {
+            ...
+        };
+
+        void hireNewManager(Manager newManager);
+    
+        Manager teamManager;
+    };
+
+    We can safely assume that the 'Manager' instance that is being passed in will be replacing the current 'teamManager' variable without looking at any other code.
+    This is because the function name and function argument clearly indicate what they are/what they do.
+
+    Your function names and parameter names should make LOGICAL SENSE.
+    Readers of your code should be able to INTUITIVELY understand what your function implementations will do without actually seeing the implementations.
+
+    Keep this in mind when you define your UDTs in this project part.
+    */
+};
 
-/*
-===================
-Part 1b Assignment: 1 Step
-===================
-1)  Fill in the 4 UDTs below with a random UDT in plain english.
-    These 4 UDTs do not need to be related.
-        a) For each plain-english UDT, write out 5 traits or 
-        properties and 3 things it can do.
-        b) remember: these properties will eventually become
-        primitives.
-        c) pick properties that can eventually be represented with 
-        'int float double bool char std::string'.
-*/
 
 /*
-Thing 1)
+Thing 1) Coffee shop
 5 properties:
-    1)
-    2)
-    3)
-    4)
-    5)
+    1) Number of employees (int)
+    2) Number of coffee maker machines (int)
+    3) Open/closed state (bool)
+    4) Name (std::string)
+    5) Date established (int)
 3 things it can do:
-    1)
-    2)
-    3)
+    1) Sell coffee
+    2) Purchase coffee beans
+    3) Toggle open/closed state
  */
 
+struct CoffeeShop
+{
+    // number of employees
+    int numEmployees = 10;
+    // number of coffee maker machines
+    int numCoffeeMachines = 2;
+    // open/closed state boolean
+    bool isOpen = true;
+    // name
+    std::string name = "The Nautical Bean";
+    // date established
+    int dateEstablished = 1995;
+
+    struct Employee
+    {
+        std::string firstName = "Jane";
+        std::string surname = "Doe";
+        int age = 20;
+        std::string role = "Cashier";
+        int id = 0;
+
+        void setRole(std::string newRole = "Generalist");
+        void setSurname(std::string newSurname);
+        bool isOlderThan(int ageToCompare = 0);   // returns true if employee age exceeds ageToCompare
+    };
+
+    Employee employeeOnDuty;
+
+    Employee getEmployeeOnDuty();  // returns the employeeOnDuty
+    void changeName(std::string newName);
+    void setEmployeeOnDuty(Employee employee);
+    void open();
+    void close();
```

Comment:
Some of these functions don't match the plain-english descriptions.
either revise the functions or change the plain-english.
- [ ] revise